cnf-certification-test/lifecycle/scaling/crd_scaling.go

```diff
@@ -0,0 +1,219 @@
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+//nolint:dupl
+
+package scaling
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/test-network-function/cnf-certification-test/internal/clientsholder"
+	"github.com/test-network-function/cnf-certification-test/pkg/tnf"
+	scalingv1 "k8s.io/api/autoscaling/v1"
+	scale "k8s.io/client-go/scale"
+
+	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	hps "k8s.io/client-go/kubernetes/typed/autoscaling/v1"
+	retry "k8s.io/client-go/util/retry"
+)
+
+//nolint:funlen
+func TestScaleCrd(hpaList []*scalingv1.HorizontalPodAutoscaler, aCrd *apiextv1.CustomResourceDefinition, timeout time.Duration) bool {
+	clients := clientsholder.GetClientsHolder()
+	var replicas int32
+	for _, version := range aCrd.Spec.Versions {
+		gvr := schema.GroupVersionResource{
+			Group:    aCrd.Spec.Group,
+			Version:  version.Name,
+			Resource: aCrd.Spec.Names.Plural,
+		}
+
+		logrus.Debugf("Looking for CRs from CRD: %s api version:%s group:%s plural:%s", aCrd.Name, version.Name, aCrd.Spec.Group, aCrd.Spec.Names.Plural)
+		crs, err := clients.DynamicClient.Resource(gvr).List(context.TODO(), metav1.ListOptions{})
+
+		if err != nil {
+			logrus.Errorf("error getting %s: %v\n", aCrd.Name, err)
+			return false
+		}
+		for _, item := range crs.Items {
+			cr := item
+			resource := schema.GroupResource{
+				Group:    aCrd.Spec.Group,
+				Resource: aCrd.Spec.Names.Plural,
+			}
+			namespace := cr.Object["metadata"].(map[string]interface{})["namespace"].(string)
+			name := cr.Object["metadata"].(map[string]interface{})["name"].(string)
+
+			scalefiled, err := clients.ScalesGetter.Scales(namespace).Get(context.TODO(), resource, name, metav1.GetOptions{})
+			if err != nil {
+				return false
+			}
+
+			tnf.ClaimFilePrintf(namespace)
+			if hpa, ok := CheckHpa(hpaList, name, namespace, aCrd.Spec.Names.Kind); ok {
+				TestScaleHPACrd(scalefiled, hpa, timeout)
+				continue
+			}
+
+			replicas = scalefiled.Spec.Replicas
+
+			if replicas <= 1 {
+				// scale up
+				replicas++
+				if !scaleCrsHelper(clients.ScalesGetter, resource, scalefiled, replicas, true) {
+					logrus.Error("can't scale cr =")
+					return false
+				}
+				// scale down
+				replicas--
+				if !scaleCrsHelper(clients.ScalesGetter, resource, scalefiled, replicas, false) {
+					logrus.Error("can't scale cr =")
+					return false
+				}
+			} else {
+				// scale down
+				replicas--
+				if !scaleCrsHelper(clients.ScalesGetter, resource, scalefiled, replicas, false) {
+					logrus.Error("can't scale cr =")
+					return false
+				} // scale up
+				replicas++
+				if !scaleCrsHelper(clients.ScalesGetter, resource, scalefiled, replicas, true) {
+					logrus.Error("can't scale cr =")
```

Comment:
Is this message incomplete? I see "can't scale cr ="

Reply:
im not sure in which commit u are looking i changed that before ? can u tell me if u still l see that incomplete?